### PR TITLE
Add user support for authentication and authorization

### PR DESCRIPTION
**Pull request by JohnNiang**
#### What type of PR is this?

/kind feature
/kind api-change

#### What this PR does / why we need it:

This PR aims to provide user support for authentication and authorization and prepares to provide ability for personal access token.

#### Which issue(s) this PR fixes:

Fixes #

##### How to test?

1. Start the application
2. See the latest log and find the password of user admin
3. Try to access <http://localhost:8090/> in the browser and you will be redirected to login page
4. Login with `admin` and `random_password_in_log`
5. Try to access:
    - <http://localhost:8090/api/v1alpha1/users>
    - <http://localhost:8090/api/v1alpha1/roles>
    - <http://localhost:8090/api/v1alpha1/rolebindings>

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Review by guqing — COMMENTED** [review]
/approve

File: src/main/java/run/halo/app/core/extension/User.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    plural = "users")
+public class User extends AbstractExtension {
+
+    @Schema(required = true)
+    private UserSpec spec;
+
+    private UserStatus status;
+
+    @Data
+    public static class UserSpec {
```
Comment (ruibaby) [inline src/main/java/run/halo/app/core/extension/User.java:33]: > **Note**
> User 的模型定义可能在后续会有改动

**Review by ruibaby — COMMENTED** [review]
/lgtm

File: src/main/java/run/halo/app/core/extension/User.java (hunk 1 above)
Comment (JohnNiang) [reply src/main/java/run/halo/app/core/extension/User.java:33]: 在未发版之前，根据需求随时变动。

**Pull request merged**